module/zfs/zvol.c

```diff
@@ -772,7 +772,7 @@ zvol_setup_zv(zvol_state_t *zv)
 	if (error)
 		return (SET_ERROR(error));
 
-	error = dnode_hold(os, ZVOL_OBJ, FTAG, &zv->zv_dn);
+	error = dnode_hold(os, ZVOL_OBJ, zv, &zv->zv_dn);
```

Comment:
... though I don't see why FTAG was wrong here.

Reply:
This was an issue because FTAG expands to the calling function name and the hold and rele are expected to provide the same tag.  So the hold was being taken with a `zvol_set_zv` tag and then released with a `zvol_shutdown_zv` tag.